spec/classes/init_spec.rb

```diff
@@ -1,1143 +1,802 @@
 require 'spec_helper'
-describe 'vmware' do
-  let(:default_facts) do
-    {
-      virtual: 'vmware',
-      vmware_has_x: false,
-      kernelrelease: '2.6.32-431.11.2.el6.x86_64',
-      os: {
-        architecture: 'x86_64',
-        family: 'RedHat',
-        name: 'RedHat',
-        release: {
-          full: '6.0',
-          major: '6',
-          minor: '0',
-        }
-      }
-    }
-  end
-  let(:facts) { default_facts }
+require 'spec_functions'
 
-  describe 'with defaults for all parameters on machine running on vmware' do
-    context 'on machine without X installed' do
-      it { is_expected.to contain_package('vmware-tools-esx-nox').with('ensure' => 'present') }
-      it { is_expected.not_to contain_package('vmware-tools-esx') }
-      it do
-        is_expected.to contain_exec('Remove vmware tools script installation').with({
-                                                                                      'command' => 'installer.sh uninstall',
-          'path'    => '/usr/bin/:/etc/vmware-tools/',
-          'onlyif'  => 'test -e "/etc/vmware-tools/locations" -a ! -e "/usr/lib/vmware-tools/dsp"',
-                                                                                    })
-      end
-      it do
-        is_expected.to contain_class('vmware::repo::redhat').only_with(
-          {
-            'repo_base_url' => 'http://packages.vmware.com/tools/esx',
-            'gpgkey_url'    => 'http://packages.vmware.com/tools/keys/VMWARE-PACKAGING-GPG-RSA-KEY.pub',
-            'esx_version'   => 'latest',
-          },
-        )
-      end
-      it do
-        is_expected.to contain_yumrepo('vmware-osps').with({
-                                                             'baseurl' => 'http://packages.vmware.com/tools/esx/latest/rhel6/x86_64',
-          'enabled'  => '1',
-          'gpgcheck' => '1',
-          'gpgkey'   => 'http://packages.vmware.com/tools/keys/VMWARE-PACKAGING-GPG-RSA-KEY.pub',
-          'proxy'    => nil,
-                                                           })
-      end
-      it do
-        is_expected.to contain_service('vmware-tools-services').with({
-                                                                       'ensure'   => 'running',
-          'require'  => 'Package[vmware-tools-esx-nox]',
-          'provider' => 'redhat',
-          'path'     => '/etc/vmware-tools/init',
-          'start'    => '/etc/vmware-tools/init/vmware-tools-services start',
-          'stop'     => '/etc/vmware-tools/init/vmware-tools-services stop',
-          'status'   => '/etc/vmware-tools/init/vmware-tools-services status',
-                                                                     })
+describe 'vmware' do
+  # By default rspec-puppet-facts only provide facts for x86_64 architectures.
+  all = {
```

Comment:
Is all of this workarounds to get 32-bit support working? Maybe we should drop it instead since it's barely used anyway? We do not usually support 32-bit in modules?

Reply:
I would love to remove 16 bit support for this module. Removing 16 bit support should remove some clutter \o/.
Can I remove 16 bit support ?

Reply:
Let's rid ourselves of this baggage!